pkg/client/metadata/metadata.go

```diff
@@ -84,11 +88,16 @@ func GetDynatraceClassicURL(ctx context.Context, client *rest.Client, environmen
 	return jsonResp.GetURL(), nil
 }
 
-func findSimpleClassicURL(ctx context.Context, client *rest.Client, environmentURL string) (url string, ok bool) {
+func findSimpleClassicURL(ctx context.Context, client client, environmentURL string) (url string, ok bool) {
 	if !featureflags.BuildSimpleClassicURL().Enabled() {
```

Comment:
was not touched in this PR, but i would move the feature flag check one level up and only call that function when it is not enabled.

Reply:
IIRC I had it outside of this function when I started the original change, then moved it in to make the calling function more readable. Honestly I'm not sure what's better or worse. The calling function does more, so outside you get a nested if (if FF on && able to build valid URL, exit) at the top of a longer function. 
As is the use of the flag is less visible which is also not great.